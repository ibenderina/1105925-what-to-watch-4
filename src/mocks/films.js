import {chooseRandomInt} from "../utils/utils";

export default [
  {
    id: 0,
    title: `Dead in a Week (Or Your Money Back)`,
    src: `https://pets.mail.ru/sharepic/68/10401.png`,
    description: `не выходи из офиса, не находи предлог,
                  зачем тебе солнце, если ты куришь «Vogue»?
                  снаружи всё злостно-хитро поэтому тут – уют,
                  не выходи из офиса, на улице точно убьют.
                  о, не выходи из офиса, не спускай взгляд с монитора,
                  потому что монитор тебе как хасиду тора,
                  и также как она абсолютно непререкаем,
                  не выходи из офиса, Авель, за дверью тебя ждёт Каин,
                  не выходи из офиса, займись квартальным отчётом,
                  если зам позвонит, не колеблясь назови идиотом,
                  зачем выходить оттуда, куда ты утром вернёшься,
                  не выходи из офиса, в аптечке имеется «Нош-па».
                  о, не выходи из офиса, отстукивай клавиатурой марши,
                  не думай о Юле, о Жанне, об Ире, о Маше;
                  с проходной веет мартом и охранника «Дошираком»,
                  не выходи из офиса, стань отшельником, но не раком.
                  не выходи из офиса, о, пускай только офис
                  знает твой истинный облик, приправленный кофе с
                  корицей, сливками, прочим; поёт мп3-соловей,
                  не выходи из офиса, на улице, чай, не Кувейт.
                  не подчиняйся общим укладам, отправь их в топку,
                  зачем по утрам познавать метро или пробку,
                  не выходи из офиса, прими шум принтера за овации,
                  спрячься от эмиссии, кризиса, безработицы, девальвации.`,
    director: `Dude`,
    starring: `Suicidal Guy`,
    genre: `Drama`,
    date: `One Million years BC`,
    background: `https://static.irk.ru/media/img/site/gallery/30414/58476b75-5f6f-42fb-97ff-9b97f79124cd_jpg_730x1000_q85.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 1,
    title: `The Gentlemen`,
    src: `https://twizz.ru/wp-content/uploads/-000//1/animals-about-to-drop-album-photos-58aeb0ce59ff1__700.jpg`,
    description: `Мой интерес, бесспорно, околофутбол. Когда играли двор на двор, я пацанам носил мячи.`,
    director: `Dude`,
    starring: `The Gentlemen`,
    genre: `Comedy`,
    date: `One Million years AC`,
    background: `https://99px.ru/sstorage/53/2017/12/tmb_217309_894087.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 2,
    title: `Lion`,
    src: `https://medialeaks.ru/wp-content/uploads/2019/08/glavnaya-1.jpg`,
    description: `Норм фильм.`,
    director: `Dude`,
    starring: `Hot Dev Patel`,
    genre: `Drama`,
    date: `A long, long time ago`,
    background: `https://img2.goodfon.ru/wallpaper/nbig/6/68/minimalizm-oblaka-blue.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 3,
    title: `Chappie`,
    src: `https://www.thepaws.net/wp-content/uploads/2018/10/funny-chihuahua.jpg`,
    description: `La-la-la`,
    director: `Dude`,
    starring: `Cute Robot`,
    genre: `Comedy`,
    date: `2015`,
    background: `https://img.desktopwallpapers.ru/flowers/pics/wide/1920x1200/fd796b086cb82fea063cd6ecafb73caf.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 4,
    title: `Guns Akimbo`,
    src: `https://memepedia.ru/wp-content/uploads/2018/12/hamster.jpg`,
    description: `Bla-bla-bla`,
    director: `Dude`,
    starring: `Harry Potter`,
    genre: `Thriller`,
    date: `Once upon a time`,
    background: `https://img5.goodfon.ru/wallpaper/big/6/80/jackson-lake-mount-moran-grand-teton-lake-ozero-gory.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 5,
    title: `Undisputed`,
    src: `https://stihi.ru/pics/2013/04/04/1852.jpg`,
    description: `Hare Krishna, everybody`,
    director: `Dude`,
    starring: `Hot fighting men`,
    genre: `Thriller`,
    date: `Not today`,
    background: `https://w-dog.ru/wallpapers/9/19/448077089475704.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 6,
    title: `En man som heter Ove`,
    src: `https://techtoday.in.ua/wp-content/uploads/2018/01/GRUMPYCAT.jpg`,
    description: `Hello`,
    director: `Dude`,
    starring: `Grouchy old man`,
    genre: `Drama`,
    date: `Recently`,
    background: `https://linchakin.com/files/word/1000/411/1.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }, {
    id: 7,
    title: `Shining`,
    src: `https://cs7.pikabu.ru/post_img/2019/03/06/7/1551870696138123277.jpg`,
    description: `The Shining is a 1980 psychological horror film produced and directed by Stanley Kubrick and co-written with novelist Diane Johnson. The film is based on Stephen King's 1977 novel of the same name and stars Jack Nicholson, Shelley Duvall, Scatman Crothers, and Danny Lloyd.`,
    director: `Dude`,
    starring: `Mad Jack Nicholson`,
    genre: `Thriller`,
    date: `666`,
    background: `https://img3.goodfon.ru/wallpaper/nbig/7/2b/priroda-risovannye-oblaka.jpg`,
    ratingScore: chooseRandomInt(0, 10),
    ratingCount: chooseRandomInt(1, 450),
  }
];
