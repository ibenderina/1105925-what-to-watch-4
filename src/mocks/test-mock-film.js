export const testMockFilm = {
  id: 1,
  title: `text`,
  src: `text`,
  description: `text`,
  director: `text`,
  starring: `text`,
  genre: `text`,
  date: `text`,
  background: `text`,
  ratingScore: 1,
  ratingCount: 1,
  url: `text`,
  comments: [{
    id: 0,
    commentAuthor: `text`,
    commentText: `text`,
    commentDate: `text`,
    commentRating: 1,
  },
  {
    id: 1,
    commentAuthor: `text`,
    commentText: `text`,
    commentDate: `text`,
    commentRating: 1,
  },
  {
    id: 2,
    commentAuthor: `text`,
    commentText: `text`,
    commentDate: `text`,
    commentRating: 1,
  }],
  runTime: `text`,
};
