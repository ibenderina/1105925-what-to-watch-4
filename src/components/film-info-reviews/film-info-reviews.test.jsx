import FilmInfoReviews from "@components/film-info-reviews/film-info-reviews";
import {testMockFilm} from "../../mocks/test-mock-film";

it(`Should FilmInfoReviews render correctly`, () => {
  const comments = testMockFilm.comments;
  let tree;

  window.act(() => {
    tree = window.create(<FilmInfoReviews
      comments={comments}
    />);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
