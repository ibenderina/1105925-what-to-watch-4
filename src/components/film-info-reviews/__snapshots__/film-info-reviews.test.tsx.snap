// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmInfoReviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Jack
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            Sun Jun 21 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9.7
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Max
          </cite>
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            Tue Jun 23 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        4
      </div>
    </div>
  </div>
</div>
`;
